### DISCUSSION TITLE
How to use v8 with generic typing?

### DISCUSSION THREAD

**Discussion by jvnm-dev** [TanStack/table#4027] [Q&A] [answered]
Hello,

I would like to make a generic table component by using `"@tanstack/react-table": "^8.0.0-beta.8"`.

Here is my code which gets a typescript error:
```tsx
import {
  createTable,
  useTableInstance,
  getCoreRowModel,
} from '@tanstack/react-table';

type TableProps<T extends object> = {
  data: T[];
  columns: {
    id: string;
    title: keyof T;
  }[];
};

export const Table = <T extends object>({ data, columns }: TableProps<T>) => {
  const table = createTable().setRowType<T>();

  const instance = useTableInstance(table, {
    data,
    columns: columns.map(column =>
      table.createDataColumn(column.title, {
        id: column.id,
        cell: info => info.getValue(),
        footer: props => props.column.id,
      }),
    ),
    getCoreRowModel: getCoreRowModel(),
  });
};
```

The error is on the second argument of `createDataColumn` and I really don't know how to get rid of it.
I also looked for similar issues but couldn't find any, so I am kindly asking for you help 🙏🏻 .

Thanks,

![image](https://user-images.githubusercontent.com/9156538/174053775-8b347dd5-fdbe-48e6-9ac6-bf61f29ea193.png)
![image](https://user-images.githubusercontent.com/9156538/174053929-d3a17195-850a-427c-9de8-ec9508b63585.png)

**Comment by mc-petry** [comment, marked as answer]
As temporary workaround try this one: 
```tsx
table.createDataColumn<AccessorFn<T>>(column.title as any, ...)
```
